### PR TITLE
added files

### PR DESCRIPTION
**Pull request by KseniaLysikova**

### CODE REVIEW

File: datasets/datasets.py
Comment (KORALLLL) [inline datasets/datasets.py:1]: разбить на два файлика. Эмбеддинг датасет лучше по-другому назвать. Он же по сути сделан для тех случаев, когда у нас одна скалярная метка для одного файла. Либо в докстринге пояснить что это за класс и где его можно использовать, либо сделать более понятное название

File: interpretability_scripts/embedding_analysis.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
+import argparse
+import os
+from utils.embeddings import get_loaders, save_metrics, save_visualization
+from models.train_models import train_emb_model
+from utils.metrics import evaluate_emb_model
+from models.models import EmbeddingCls
```
Comment (KORALLLL) [inline interpretability_scripts/embedding_analysis.py:6]: это не будет работать, пофиксить

File: interpretability_scripts/probing.py
Comment (KORALLLL) [inline interpretability_scripts/probing.py:1]: Всё это полотнище выглядит сложно, упростить и разбить на отдельные понятные функции

File: models/models.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+class EmbeddingCls(nn.Module):
+    """
+    Baseline model class for embeddings gender classification
+    """
+
+    def __init__(self, input_dim=256, num_classes=2):
+        super(EmbeddingCls, self).__init__()
+        self.fc1 = nn.Linear(input_dim, 128)
+        self.fc2 = nn.Linear(128, num_classes)
```
Comment (KORALLLL) [inline models/models.py:12]: Сделать возможность задавать hidden dim

File: models/models.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import torch.nn as nn
+
+
+class EmbeddingCls(nn.Module):
+    """
+    Baseline model class for embeddings gender classification
+    """
```
Comment (KORALLLL) [inline models/models.py:7]: не совсем для генеде классификации. По сути эта модель будет подходить и для регрессии. Лучше просто написать, что базовый лкасс для модели

File: models/models.py
Comment (KORALLLL) [inline models/models.py:1]: каждую модель в свою папку закинуть

File: datasets/datasets.py [hunk 4]
```diff
@@ -0,0 +1,102 @@
+from base_dataset import BaseDataset
```
Comment (KORALLLL) [inline datasets/datasets.py:1]: это скорее всего не будет работать

File: interpretability_scripts/probing.py [hunk 5]
```diff
@@ -0,0 +1,257 @@
...
+from utils.metrics import (
+    evaluate_probing,
+    read_metrics,
+    plot_metrics,
+    save_metrics,
+    save_to_csv
+)
+from datasets.datasets import ActivationDataset
+from models.models import ProbingCls
+from models.train_models import train_probing_model
```
Comment (KORALLLL) [inline interpretability_scripts/probing.py:19]: не будет работать скорее всего, пофиксить

File: models/train_models.py [hunk 6]
```diff
@@ -0,0 +1,61 @@
+from models import ProbingCls
```
Comment (KORALLLL) [inline models/train_models.py:1]: не будет работать скорее всего

File: utils/embeddings.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
+import os
+from datasets.datasets import EmbeddingsDataset
```
Comment (KORALLLL) [inline utils/embeddings.py:2]: скорее всего не будет работать

File: utils/layers.py [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+    def forward(
+            self, x, target_layer, from_activation=False, identity_file=None
+    ):
+        activations = {}
+        model_front = self.model.model.front
+        out = x
+        if not from_activation:
+            out = x.permute(0, 2, 1).unsqueeze(dim=1)
+            out = model_front.relu(model_front.bn1(model_front.conv1(out)))
```
Comment (KORALLLL) [inline utils/layers.py:42]: разбей на подфункции, выглядит массивно я тяжело для восприятия

**Review by KORALLLL — CHANGES_REQUESTED** [review]

**Review by KORALLLL — CHANGES_REQUESTED** [review]

**Review by KORALLLL — APPROVED** [review]
